corehq/apps/users/forms.py

```diff
@@ -379,6 +379,110 @@ def clean(self):
     validate_username = EmailValidator(message=ugettext_lazy(u'Username contains invalid characters.'))
 
 
+_username_help = """
+<span ng-if="usernameAvailabilityStatus === 'pending'">
+    <i class="fa fa-circle-o-notch fa-spin"></i>
+    %(checking)s
+</span>
+<span ng-if="usernameAvailabilityStatus === 'taken'"
+      style="word-wrap:break-word;">
+    <i class="fa fa-remove"></i>
+    {{ usernameStatusMessage }}
+</span>
+<span ng-if="usernameAvailabilityStatus === 'available'"
+      style="word-wrap:break-word;">
+    <i class="fa fa-check"></i>
+    {{ usernameStatusMessage }}
+</span>
+<span ng-if="usernameAvailabilityStatus === 'error'">
+    <i class="fa fa-exclamation-triangle"></i>
+    %(server_error)s
+</span>
+""" % {
+    'checking': ugettext_noop('Checking Availability...'),
+    'server_error': ugettext_noop('Issue connecting to server. Check Internet connection.')
+}
```

Comment:
http://django-angular.readthedocs.org/en/latest/integration.html#template-tags

"The cleanest solution to circumvent this, is by using the verbatim tag, which became available in Django 1.5." (angularjs tag is even better)

"This approach is less verbose than using the verbatim tag. The problem, however, is that you have to remember to use this alternative tag syntax for all of your AngularJS templates. It also makes it difficult to integrate third party AngularJS directives, which are shipped with their own templates."

It also makes tests really complicated.